src/components/ResourceItem/tests/ResourceItem.test.tsx

```diff
@@ -310,106 +321,125 @@ describe('<ResourceItem />', () => {
+        .find('div', {className: styles.ResourceItem})!
+        .trigger('onClick', {
+          stopPropagation: () => {},
+          nativeEvent: {},
+        });
       expect(onClick).toHaveBeenCalledWith(itemId);
     });
 
     it('calls window.open on metaKey + click', () => {
-      const wrapper = mountWithAppProvider(
+      const wrapper = mountWithApp(
         <ResourceListContext.Provider value={mockDefaultContext}>
           <ResourceItem id={itemId} url={url} accessibilityLabel={ariaLabel} />
         </ResourceListContext.Provider>,
       );
-      const item = findByTestID(wrapper, 'Item-Wrapper');
-      trigger(item, 'onClick', {
-        stopPropagation: () => {},
-        nativeEvent: {metaKey: true},
-      });
+
+      wrapper
+        .find('div', {className: styles.ResourceItem})!
+        .trigger('onClick', {
+          stopPropagation: () => {},
+          nativeEvent: {metaKey: true},
+        });
       expect(spy).toHaveBeenCalledWith(url, '_blank');
     });
 
     it('calls onClick when hitting keyUp on the item when onClick and URL exists', () => {
       const onClick = jest.fn();
-      const wrapper = mountWithAppProvider(
+      const wrapper = mountWithApp(
         <ResourceListContext.Provider value={mockDefaultContext}>
           <ResourceItem id={itemId} url="#" onClick={onClick} />
         </ResourceListContext.Provider>,
       );
 
-      findByTestID(wrapper, 'Item-Wrapper').simulate('keyup', {
-        key: 'Enter',
-      });
+      wrapper
+        .find('div', {className: styles.ResourceItem})!
+        .trigger('onKeyUp', {key: 'Enter'});
 
       expect(onClick).toHaveBeenCalled();
     });
 
     it('does not call onClick when hitting keyUp on non Enter key', () => {
       const onClick = jest.fn();
-      const wrapper = mountWithAppProvider(
+      const wrapper = mountWithApp(
         <ResourceListContext.Provider value={mockDefaultContext}>
           <ResourceItem id={itemId} url="#" onClick={onClick} />
         </ResourceListContext.Provider>,
       );
 
-      findByTestID(wrapper, 'Item-Wrapper').simulate('keyup', {
-        key: 'Tab',
-      });
+      wrapper
+        .find('div', {className: styles.ResourceItem})!
+        .trigger('onKeyUp', {key: 'Tab'});
 
       expect(onClick).not.toHaveBeenCalled();
     });
 
     it('does not call onClick when hitting keyUp on the item when no URL exists', () => {
       const onClick = jest.fn();
-      const wrapper = mountWithAppProvider(
+      const wrapper = mountWithApp(
         <ResourceListContext.Provider value={mockSelectModeContext}>
           <ResourceItem id={itemId} onClick={onClick} />
         </ResourceListContext.Provider>,
       );
 
-      findByTestID(wrapper, 'Item-Wrapper').simulate('keyup', {
-        key: 'Enter',
-      });
+      wrapper
+        .find('div', {
+          className: classNames(
+            styles.ResourceItem,
+            styles.selectable,
+            styles.selectMode,
```

Comment:
Yeah, I don't like it either. On top of the reasons you've mentioned, it kinda looks brittle to me (and sometimes complex to figure out which classes are expected depending on which props were passed to the component). I was hoping you folks would be able to suggest me a better alternative. Should I leave it like that, then?

Reply:
Slow down will be negligible. The style file is already part of the module graph because ResourceItem itself imports it too. When Jest encounters .scss files it basically maps in `identity-obj-proxy` which is some very trivial JS that says "any key is valid, return that value as a string" - so `styles.ResourceItem` is `'ResourceItem'`, `styles.selectable` is `'selectable'` etc.


Agree with Kyle, that the while not super great it's better than keeping using `testID` stuff, and I've got not better idea.

To reduce the need to play "guess all the classes' whachamole I think you might be able to use `expect.stringContaining` to reference just the one

```js
wrapper .find('div', {
  className: expect.stringContaining(styles.ResourceItem)
})!.trigger('onKeyUp', {key: 'Enter'})
```


e.g. https://github.com/Shopify/polaris-react/blob/746d2db42967b8a92a6a291860e18be2c9cc05da/src/components/Button/tests/Button.test.tsx#L424-L426

Reply:
Thanks, everyone! 

As @BPScott had anticipated, the `expect.stringContaining` only works within the context of jest matchers, but I was able to introduce it to a few places anyway. I've used the `findWhere` everywhere else.

I've also incorporated a bit of @AndrewMusgrave 's suggestion to extract the logic into a single place, hopefully it's looking better now. 

Thanks @kyledurand for spotting the places I've missed! 

There are now just these 2 points before I can consider this done. Github won't let me paste the direct links to the discussions no matter what I try (it keeps removing the anchor from the link), so please copy the following anchors and paste it at the end of this PR's URL to go straight to them: **#discussion_r683666010** and **#discussion_r686332281**